packages/validator/src/services/syncingStatusTracker.ts

```diff
@@ -0,0 +1,60 @@
+import {ApiClient, routes} from "@lodestar/api";
+import {Logger} from "@lodestar/utils";
+import {Slot} from "@lodestar/types";
+import {IClock} from "../util/clock.js";
+
+export type SyncingStatus = routes.node.SyncingStatus;
+
+type RunOnSyncedFn = (slot: Slot, signal: AbortSignal) => Promise<void>;
+
+/**
+ * Track the syncing status of connected beacon node
+ */
+export class SyncingStatusTracker {
+  private prevSyncingStatus: SyncingStatus | null = null;
+
+  private readonly fns: RunOnSyncedFn[] = [];
+
+  constructor(
+    private readonly logger: Logger,
+    private readonly api: ApiClient,
+    private readonly clock: IClock
+  ) {
+    this.clock.runEverySlot(this.checkSyncingStatus);
+  }
+
+  /** Run function when node status changes from syncing to synced */
+  runOnSynced(fn: RunOnSyncedFn): void {
+    this.fns.push(fn);
+  }
+
+  private checkSyncingStatus = async (slot: Slot, signal: AbortSignal): Promise<void> => {
+    try {
+      const syncingStatus = (await this.api.node.getSyncingStatus()).value();
+
+      if (this.prevSyncingStatus === null) {
+        // Initial check, do not run any handlers
+        this.prevSyncingStatus = syncingStatus;
+      } else {
+        if (this.prevSyncingStatus.isSyncing === true && syncingStatus.isSyncing === false) {
+          for (const fn of this.fns) {
+            fn(slot, signal).catch((e) => this.logger.error("Error calling sync event handler", e));
+          }
+        }
+        this.prevSyncingStatus = syncingStatus;
+      }
+
+      if (syncingStatus.isSyncing === true) {
+        this.logger.warn("Connected beacon node is syncing", {slot, ...syncingStatus});
+      } else {
+        this.logger.debug("Connected beacon node is synced", {slot, ...syncingStatus});
+      }
+    } catch (e) {
+      this.logger.error("Failed to check syncing status", {}, e as Error);
```

Comment:
what's the UX of this (and can it be improved)? Most of the time we'll get two logs here, right? both health log and this one?

Reply:
So we shouldn't have two logs, that seems wrong, need to review

I think a good UX would be
- `verbose`: once a slot to print sync status if `isSyncing=false`
- `warn`: once a slot if `isSyncing=true`
- `error`: once a slot if failed to check status (e.g. due to node offline)

but in any case, just one of those, never multiple logs per slot

Reply:
> both health log and this one?

ah you mean a log related to polling health status api? There is no log like this as far as I know

Reply:
The remaining ux question for me right now is how to deal with existing "Node is syncing" logs which are printed out due to other api failures, like polling duties and are already throttled to once per slot.

I aligned the logs to be somewhat similar
![image](https://github.com/user-attachments/assets/95c1044f-89ce-44da-928e-b25f15adae6f)

but it might look a bit redundant because `getProposerDuties` is called every slot right now. I am still not sure if removing those altogether is good because it still provides information about polling failures + includes error message of beacon node.

Reply:
I think the UX is clean now
- `info` log to inform user about sync status just once on startup or on resynced
- `warn` every slot if not synced with details about head slot / sync distance
- `verbose` dump node syncing status very slot, including all data retrieved from api

There might be additional "Node is syncing" warnings due to duty polling attempts running into 503 errors but those are slightly different in nature as (1) they include error details (hence important) and (2) depend on lifecycle of imported keys and might not even fire but I think it's good to inform the user about the node sync status in any case (think pending validator).

Reply:
Could further consider the case if beacon node is synced but connected EL client is syncing or offline and emit a warning every slot. Might be out of scope though since status tracker service is primarily designed to trigger duties polling.